packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts

```diff
@@ -2252,6 +2253,87 @@ integTest(
   }),
 );
 
+/**
+ * Create an S3 bucket, orphan that bucket, then import the bucket, with a NodeJSFunction lambda also in the stack.
+ *
+ * Validates fix for https://github.com/aws/aws-cdk/issues/31999 (import fails)
+ */
+integTest(
+  'test s3 bucket import with nodejsfunction lambda',
+  withDefaultFixture(async (fixture) => {
+    // Create a temporary directory for esbuild installation
+    const tempDir = mkdtempSync(path.join(os.tmpdir(), 'esbuild-'));
+    const originalPath = process.env.PATH;
+
+    // Install esbuild locally in the temporary directory
+    execSync('npm init -y && npm install esbuild', { cwd: tempDir, stdio: 'inherit' });
```

Comment:
Hmm, this shouldn't be necessary...we have `esbuild` a bunch in our `devDependencies` so it should already be installed.

> For example: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/package.json#L175

See if you can without it, maybe its just a matter of locating the installed one and adding it to the path, but I believe that shouldn't be necessary as well.

Reply:
`esbuild` is needed as a runtime dependency to use it in integration tests, but we only keep it as a dev dependency in some packages. We do not make it available as a runtime dependency because it is not available via JSII, according to @mrgrain. We don't have it as a dependency in `@aws-cdk-testing`, in either `framework-integ` or `cli-integ` `package.json`s. I was able to remove the code above, though, by forcing the integ test to use Docker `esbuild` to bundle the code instead.